hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestPayloadDeprecationFlow.scala

```diff
@@ -597,11 +605,18 @@ class TestPayloadDeprecationFlow extends SparkClientFunctionalTestHarness {
       "Data should remain consistent after downgrade including new row")
   }
 
-  def getExpectedResultForSnapshotQuery(payloadClazz: String, usesDeleteMarker: Boolean): Seq[(Int, Long, String, String, Double, String, String, Int, Int, String)] = {
+  def getExpectedResultForSnapshotQuery(payloadClazz: String, usesDeleteMarker: Boolean, hasOrderingField: Boolean, tableType: String, startVersion: String = "8"): Seq[(Int, Long, String, String, Double, String, String, Int, Int, String)] = {
+    // V6 backward compatibility: OverwriteWithLatestAvroPayload with MOR + ordering field uses EVENT_TIME
+    val isV6BackwardCompatCase = startVersion == "6" &&
```

Comment:
will remove this now as we are not covering this case in this test, and will be doing a different approach